types/types.proto

```diff
@@ -214,38 +215,50 @@ message BlockGossip {
 //----------------------------------------
 // Blockchain Types
 
+// just the minimum the app might need
 message Header {
+  // basics
   string chain_id = 1 [(gogoproto.customname)="ChainID"];
   int64 height = 2;
   int64 time = 3;
+
+  // txs
   int32 num_txs = 4;
-  BlockID last_block_id = 5 [(gogoproto.customname)="LastBlockID", (gogoproto.nullable)=false];
-  bytes last_commit_hash = 6;
-  bytes data_hash = 7;
-  bytes validators_hash = 8;
-  bytes app_hash = 9;
+  int64 total_txs = 5;
```

Comment:
what's the difference between num_txs and total_txs?

Reply:
I think num_tx is the number of transaction in this block, but total_tx is total transaction since genesis time.

Reply:
right. I forgot. what happens when total_txs overflows? @ebuchman

Reply:
It should be end of the world! 2 in power of 64 transactions!